### ISSUE TITLE
Support for ExpressionStatements 

### ISSUE THREAD

**Issue by gregjacobs** [dsherret/ts-simple-ast#164]
Hey @dsherret 

I've been going through more of your project today, and first off: excellent work. This project is huge, and very impressive. 👍 

I'm already using `ts-simple-ast` for some transformations of source code for a few scenarios (I've mentioned transforming Swagger generated code in one of your other issues), but now I'm working on my holy grail: Converting our Angular 1.x codebase to Angular 2+. (Major project, but I'm starting small!)

What I want to do is take a source file and loop over all of the statements in it, understand them, and then write a new file. To give you an example, a simple example source file might look like this (although there might be more statements in it):

```js
var userUrl = "/path/to/user";

angular.module( 'ourApp' ).factory( 'UserService', [ '$http', function( $http ) {
    return {
        getUser: function getUser() {
            return $http.get( userUrl );
        }
    };
} ] );
```

If I'm looping over the statements in this source file though, the `angular.module` line is an `ExpressionStatement`. I noticed that there are no methods in `TypeGuards` to check for an `ExpressionStatement`, and no wrapper for this node type yet either. Can this be added?

I can attempt a PR too if this is the right approach to add these things

**Comment by gregjacobs** [comment]
Btw, just as an fyi, I would ideally convert the above AngularJS code to something like this:

```ts
import { Injectable } from '@angular/core';
import { Http } from '@angular/http';

@Injectable()
export class UserService {
    private userUrl = "/path/to/user";

    constructor( private http: Http ) {}

    getUser() {
        return this.http.get( this.userUrl ).toPromise();
    }
}
```

**Comment by dsherret** [comment]
Nice! I don't know a lot about angular, but that does sound challenging.

This is implemented in 3.1.0.

By the way, in the future, please continue to open an issue like this one, but note that you can always navigate nodes using the getChild-like methods and can tell if something is a certain node by looking at the result of the `.getKind()` method.

For example, pre 3.1.0 you could do this:

```ts
import * as ts from "typescript";
import Ast from "ts-simple-ast";

const sourceFile = ...;

for (const statement of sourceFile.getStatements()) {
    if (statement.getKind() === ts.SyntaxKind.ExpressionStatement) {
        // expresison statement will only have one child
        const expression = statement.getChildren()[0];
    }
}
```

But now in 3.1.0 you can just do this:

```ts
import * as ts from "typescript";
import Ast, {TypeGuards} from "ts-simple-ast";

const sourceFile = ...;

for (const statement of sourceFile.getStatements()) {
    if (TypeGuards.isExpressionStatement(statement)) {
        const expression = statement.getExpression();
    }
}
```

**Comment by gregjacobs** [comment]
Awesomeness, thanks for adding! 

I did eventually use `.getKind()` last night, but will definitely be nice to have a `ts-simple-ast`'ized way of doing things :) 

The example you posted here might be good in the docs btw in the navigation section. Did take me a bit to find that there was a `getStatements()` method. I decided to use `getStatements()` over some of the query-like methods btw because I want my script to throw an error for anything it comes across that it doesn't know how to handle yet.
